pkg/config/application.go

```diff
@@ -64,6 +64,9 @@ type DeploymentPlanner struct {
 	// Disable auto-detecting to use QUICK_SYNC or PROGRESSIVE_SYNC.
 	// Always use the speficied pipeline for all deployments.
 	AlwaysUsePipeline bool `json:"alwaysUsePipeline"`
+	// Automatically reverts all deployment changes on failure.
+	// Default is true.
+	AutoRollback *bool `json:"autoRollback,omitempty" default:"true"`
```

Comment:
[imo] How about adding the comment to explain this is only for the v1 for now? It's because we avoid using it on the current version.

Reply:
I think we can add a deprecated note to the `Input.AutoRollback` 👍

Reply:
@ffjlabo Done by https://github.com/pipe-cd/pipecd/pull/5052/commits/ebe5f5f786717c793a64cab0ac7a008e09cbfce8